experimental/listener.go

```diff
@@ -13,14 +13,22 @@ import (
 // iteration. As a result, parameter values may be different than the ones their
 // function was called with.
 type StackIterator interface {
-	// Next moves the iterator to the next function in the stack. Returns false
-	// if it reached the bottom of the stack.
+	// Next moves the iterator to the next function in the stack. Returns
+	// false if it reached the bottom of the stack.
 	Next() bool
 	// FunctionDefinition returns the function type of the current function.
 	FunctionDefinition() api.FunctionDefinition
+	// SourceOffset computes the offset in the module Code section where the
+	// call occured (translated for native functions), or the beginning of
+	// function for the top of the stack. Returns 0 if the source offset
```

Comment:
@ncruces have any 👨‍🎨 on the -1 vs zero thing for source offset not available. I know we recently chatted on tinygo (and otherwise) using zero as nil though technically wasm can start from zero.

I think basically nothing will start at line zero in the source file, is why we can use zero here. (it isn't about the binary rather the %.c %.rs etc "the source"

Reply:
Yes, I don't think what we chatted about applies here.

TLDR it was: linear memory address zero in C (and probably TinyGo) can't be dereferenced, so I choose to panic when the host gets a zero pointer it's expected to dereference (although the host _can_ read/write address zero).

Reply:
As long as 0 is not a valid offset and given this is unsigned, zero seams reasonable.

Reply:
apples and oranges, but what's common (fruit?) is we are banishing a zero offset which isn't usually valid.

So, the simile is like memory offset zero is almost always invalid, source offset zero would be pretty invalid.

nerdy side bar:

As far as I can think of, the only languages which could use either are %.wat and an edge case of that, also. For example, text format can use memory offset zero, and also strangely, you can omit `module` and have a function as a top-level element in the file.. otoh I doubt if source-mapped you'd end up with index zero as it would not be in the body part ;) https://www.w3.org/TR/wasm-core-1/#abbreviations%E2%91%A0%E2%91%A5